### PR TITLE
Health checks lost fail task

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

With this PR: https://github.com/confluentinc/cp-ansible/pull/642 the feature where a health check failure both exported logs **and** failed an installation was lost. Need to make sure the playbook stops when a health check fails.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Force a failure with a bad custom property

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.control_center/tasks/health_check.yml [hunk 1]
```diff
@@ -28,3 +28,8 @@
     group: "{{control_center_group}}"
   # Only one health check runs, only one will have a 'failed' field. For skipped checks, defaulting 'failed' to False
   when: result1.failed|default(False)
+
+- name: Fail Provisioning
+  fail:
```
Comment (domenicbove) [inline roles/confluent.control_center/tasks/health_check.yml:33]: @erikgb any idea whats wrong with this task linting wise?
https://ansible-lint.readthedocs.io/en/latest/default_rules.html#ignore-errors

Am i not allowed to use the fail module?

Comment (erikgb) [reply roles/confluent.control_center/tasks/health_check.yml:33]: This is the error message:

> 35:38     error    no new line character at the end of file  (new-line-at-end-of-file)

At least now. 😄

Comment (domenicbove) [reply roles/confluent.control_center/tasks/health_check.yml:33]: Oh duh! I was reading the ansible-lint logs. Thanks

**Comment by erikgb** [comment]
A suggestion: Use a block to avoid repeating the failed condition?

**Review by erikgb — APPROVED** [review]
LGTM

**Comment by domenicbove** [comment]
@erikgb I like the block, but ansible-lint doesn't...
```
dbove@C02DV0YAMD6T cp-ansible % ansible-lint -c .ansible-lint -qq --force-color *.yml
Loading custom .yamllint config file, this extends our internal yamllint config.
Traceback (most recent call last):
  File "/Users/dbove/Library/Python/3.8/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansiblelint/__main__.py", line 279, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansiblelint/__main__.py", line 216, in main
    result = _get_matches(rules, options)
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansiblelint/runner.py", line 197, in _get_matches
    matches.extend(runner.run())
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansiblelint/runner.py", line 132, in run
    matches.extend(self._emit_matches(files))
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansiblelint/runner.py", line 165, in _emit_matches
    for child in ansiblelint.utils.find_children(lintable):
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansiblelint/utils.py", line 250, in find_children
    for token in split_args(path_str):
  File "/Users/dbove/Library/Python/3.8/lib/python/site-packages/ansible/parsing/splitter.py", line 285, in split_args
    raise AnsibleParserError(u"failed at splitting arguments, either an unbalanced jinja2 block or quotes: {0}".format(args))
ansible.errors.AnsibleParserError: failed at splitting arguments, either an unbalanced jinja2 block or quotes: roles/confluent.kafka_rest/tasks/confluent.common/tasks/fetch_logs.yml'}
```
WDYT? It doesnt feel right to exclude these files

**Comment by erikgb** [comment]
The path in the error message gives a hint: `roles/confluent.kafka_rest/tasks/confluent.common/tasks/fetch_logs.yml`. The problem might be that the relative pathing somehow breaks when putting the task inside a block. The task in question looks like this:

```code
- name: Fetch Files for Debugging Failure
  # Cannot use include_role in Ansible Handlers: https://github.com/ansible/ansible/issues/20493
  include_tasks:
    file: ../../confluent.common/tasks/fetch_logs.yml
```

It seems like a bug in Ansible Lint. Seems like the only option is to ignore the problematic files, at least for now. Maybe add a comment to the ignore statement? I can check with ansible-lint maintainer(s).

**Comment by domenicbove** [comment]
@erikgb Looks like github actions minutes ran out... We lost the lint checks

**Comment by erikgb** [comment]
> @erikgb Looks like github actions minutes ran out... We lost the lint checks

@domenicbove That's sad. Are you able to find a sponsor (hint, hint)? 😉

**Comment by erikgb** [comment]
I found a probably related ansible-lint issue: https://github.com/ansible-community/ansible-lint/issues/476

**Pull request merged**